### ISSUE TITLE
New Docs pages

### ISSUE THREAD

**Issue by hzoo** [babel/website#1628]
In addition to the docs for v7 release, we should revamp a lot of missing information. Creating this issue to track some high level and specific topics to add and people can take them as we go.

- [ ] Config (currently https://github.com/babel/website/blob/master/docs/babelrc.md)
  - lookup behavior
  - file names (`.babelrc` vs. `package.json`, etc) and why you would use which, and examples
  - use cases: how it is used with compiling `node_modules`, symlinking, monorepo, node, webpack
  - commonly used options: `env` and `overrides`

**Comment by dczombera** [comment]
I think #334 is particular important for people who want to get started writing plugins and/or want to contribute to the project.
Even though I have only touched the tip of the iceberg when it comes to understanding the project in order to be able to contribute more serious stuff, I found it very tough to even get started because of the lack of documentation for the API. I think people who consider helping out or want to write plugins are easily scared away and stop all of their efforts because they are frustrated.

I really would like to help you out here because I think it is important and it would also help me to improve my understanding. The question I'm asking myself right now where to start here? :D 🤔

**Comment by hzoo** [comment]
@dczombera I totally agree! Problem is that everything _feels_ important, and I like having docs for usage first is more important. It's also more likely that you can figure it out since we've never really had API docs so I would prioritize other docs first to be practical given limited time/resources. I would hope that every person that has learned how to make a plugin would be able to contribute to these docs. And yes that was under James' account so we'd like to move that into the new site in addition to translations, was always the plan.

> I found it very tough to even get started

I'd hope being available (maybe too much) on Slack/Twitter would help here. Yeah the issue is that there's a lot of things to be working on and I would love to improve on this now.

**Comment by sw-yx** [comment]
hello! doing some work on the handbook today. i will post a WIP PR.

**Comment by lovetingyuan** [comment]
I hope all the `babel-traverse` and `babel-types` API cound be added to the new doc site, it would be a big deal to those want to develope plugins

**Comment by sw-yx** [comment]
i would love to work on that once the handbook is merged in. anyone i can work with on this?
